drivers/lora/Kconfig.sx12xx

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2019 Manivannan Sadhasivam
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+DT_COMPAT_SX1276 = semtech,sx1276
+DT_COMPAT_SX1261 = semtech,sx1261
+DT_COMPAT_SX1262 = semtech,sx1262
+
+menuconfig LORA_SX12XX
+	bool "Semtech SX-series driver"
+	select HAS_SEMTECH_RADIO_DRIVERS
```

Comment:
Why are these selects necessary? Isn't `#ifdef CONFIG_LORA_SX12XX` enough in software?

Usually, `HAS_FOO` macros are used to signal the presence of some hardware, or when an arch supports a feature, or something like that. This seems a bit strange.

Reply:
They are used to control the build of the LoRaMAC-Node library. The `CONFIG_HAS_LORAMAC` macro includes the MAC and the `CONFIG_HAS_SEMTECH_SX*` macros select which radio to build in the library.

Not sure what the rationale was since these were used by the original SX1276 driver.

Reply:
If this is just moving existing options around, OK, but this looks like unconventional usage. Thanks.